From c7e4a012be955e2a64079428b0230c8b1014e66f Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Fri, 10 Jun 2022 14:11:03 +0300
Subject: [PATCH] [SW-70527] gaudi2: ras: mme reset after bist

In order to avoid ECC after running BIST on MME rings,
support added to perform MME unit reset.

Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
Change-Id: Ib86ea7dcb4bc77f6b189ec48de30faaaad2e3446
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index a6b3d35..5c6fbfe 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -1086,6 +1086,7 @@
 	regmap_writel(PSOC_RESET_CONF_ROTATOR_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_ROTATOR_UNIT_RST);
 	regmap_writel(PSOC_RESET_CONF_SM_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_SM_UNIT_RST);
 	regmap_writel(PSOC_RESET_CONF_HBM_MC_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_HBM_MC_UNIT_RST);
+	regmap_writel(PSOC_RESET_CONF_MME_EUS_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_MME_EUS_UNIT_RST);
 	regmap_writel(PSOC_RESET_CONF_UNIT_RST_N_IND_MASK, mmPSOC_RESET_CONF_UNIT_RST_N);
 	udelay(DELAY_1K_USEC); /* TODO: confirm the delay */
 	regmap_writel(0, mmPSOC_RESET_CONF_UNIT_RST_N);
@@ -1094,6 +1095,7 @@
 	regmap_writel(0, mmPSOC_RESET_CONF_SM_UNIT_RST);
 	regmap_writel(0, mmPSOC_RESET_CONF_ROTATOR_UNIT_RST);
 	regmap_writel(0, mmPSOC_RESET_CONF_HBM_MC_UNIT_RST);
+	regmap_writel(0, mmPSOC_RESET_CONF_MME_EUS_UNIT_RST);
 	udelay(DELAY_1K_USEC); /* TODO: confirm the delay */
 
 	/* Restore original iso value */
